### PR TITLE
Add customer.io email-change primitives (unused, tested)

### PR DESCRIPTION
**Pull request by Will-Howard**
# Description

Part of the work to allow users to change their email (#2570), specifically making customer.io handle this correctly (#2810).

This PR adds `updateCustomerIoEmail({ userId, oldEmail, newEmail })`, which tries to keep a stable customer.io profile pinned to the userId, and change the email from `oldEmail` to `newEmail`. There are edge cases where e.g. the `newEmail` already exists in customer.io (which can happen if a user applied to ~any BlueDot thing under the new email already), it's best to look at the tests to see how that is handled in each case.

I also added `sendEmailChangeVerification` in this PR, in order to meaningfully test more of `updateCustomerIoEmail`, but `sendEmailChangeVerification` isn't yet used by anything in this PR. It will be adopted by the actual email-change flow when that exists.

I ran some tests against the prod environment in preparing this branch, per the comment in the test file you can review these tests here:
> These semantics were validated against the real production workspace with a live-fire
> model-based test, which is preserved on the `wh-2810-cio-livetest-2026-07-archive` git branch,
> file `apps/website/src/lib/api/customerio.prod.test.ts`.
I'm not committing these tests themselves because they're not practical to maintain (they create and delete real users in customer.io every time).

## Issue

#2810

## Developer checklist

- [x] N/A Front-end code follows [Component Accessibility Checklist (for Testing)](https://github.com/bluedotimpact/bluedot/blob/master/DEVELOPMENT_HANDBOOK.md#component-accessibility-checklist)
- [x] Considered having snapshot tests and/or happy path functional tests
- [x] N/A Considered adding Storybook stories

### CODE REVIEW

**Comment by Will-Howard** [comment]
@coderabbitai review @greptileai review

**Comment by Will-Howard** [comment]
> Up to ~60s of synchronous blocking per call, once wired into a request path.

On this high level comment: This is intended for use as a non-blocking side effect after email change succeeds, so it's ok for it to take some time

File: apps/website/src/lib/api/customerio.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  if (!res.ok) throw new Error(`customer.io identify failed: HTTP ${res.status}`);
+}
+
+const profileEmail = (profile: CioProfile | null): string | null => {
+  const email = profile?.identifiers?.email ?? profile?.attributes?.email;
+  return typeof email === 'string' ? normaliseEmail(email) : null;
+};
+
+async function emailIsVisible({ userId, email }: { userId: string; email: string }): Promise<boolean> {
+  for (let attempt = 0; attempt < VERIFY_POLL_ATTEMPTS; attempt++) {
```
Comment (marn-in-prod) [inline apps/website/src/lib/api/customerio.ts:69]: Nit: the loop retries when the write isn't possible, but not when getProfileById throws. Is it intentional that a single throw will stop the loop?

File: apps/website/src/lib/api/customerio.test.ts [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+};
+
+const makeFakeCio = (profiles: FakeProfile[], { applyWrites = true }: { applyWrites?: boolean } = {}): FakeCio => ({
+  profiles,
+  applyWrites,
+  identifyCalls: [],
+});
+
+const applyIdentify = (cio: FakeCio, id: string, email: string) => {
+  const owner = cio.profiles.find((p) => p.email === email);
```
Comment (marn-in-prod) [inline apps/website/src/lib/api/customerio.test.ts]: Nit: If cio is email case-insensitive IRL then we should make the test also behave that way, right? Plus we should also add a test where the stored email isn't all lowercase

**Review by marn-in-prod — APPROVED** [review]
Generally looks good! Comments are about edge cases and me wanting clarity on what you intended

File: apps/website/src/lib/api/customerio.ts (hunk 1 above)
Comment (Will-Howard) [reply apps/website/src/lib/api/customerio.ts:69]: > Is it intentional that a single throw will stop the loop?

Not intentional, I'll add a `catch` in here. It should be safe to retry `getProfileById` since it's just a read

File: apps/website/src/lib/api/customerio.test.ts (hunk 2 above)
Comment (Will-Howard) [reply apps/website/src/lib/api/customerio.test.ts]: I've updated `applyIdentify` to do everything case-insensitively, and added a test before for the case where the email stored on the user is in mixed case.

I also checked how CIO itself works, and it converts the email to lower case before writing it (so any email you read back will always be lowercase), the tests are now consistent with that
